src/test/java/ArrayIndexedBagTest.java

```diff
@@ -0,0 +1,258 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ArrayIndexedBagTest {
+
+    @Test
+    void aNewIndexedBagIsEmpty() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        assertTrue(bag.isEmpty());
+    }
+
+    @Test
+    void emptyIndexedBagHasSizeZero() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        assertEquals(0, bag.size());
+    }
+
+    @Test
+    void nonEmptyIndexedBagIsEmptyReturnsFalse() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(99);
+        assertFalse(bag.isEmpty());
+    }
+
+    @Test
+    void addingToEndOfIndexedBagUpdatesSize() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        // Use both versions of add
+        bag.add(99);
+        bag.add(101);
+        assertEquals(2, bag.size());
+    }
+
+    @Test
+    void addingToIndexOfIndexedBagUpdatesSize() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        // Use both versions of add
+        bag.add(99);
+        bag.add(0, 101);
+        assertEquals(2, bag.size());
+    }
+
+    @Test
+    void containsAddedElementToIndexedBagReturnsTrue() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(99);
+        assertTrue(bag.contains(99));
+    }
+
+    @Test
+    void containsNonAddedElementToIndexedBagReturnsFalse() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(99);
+        assertFalse(bag.contains(101));
+    }
```

Comment:
Could also include a test case for `contains` returning `true` when there are multiple of that element.

Reply:
:rabbit: